reinforcement_learning/bindings/cs/rl.net.cli/EntryPoints.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.IO;
+using Rl.Net;
+
+namespace Rl.Net.Cli {
+    static class EntryPoints
+    {
+        public static void Main(string [] args)
+        {
+            BasicUsageExample(args);
+        }
+
+        private static void WriteErrorAndExit(string errorMessage, int exitCode = -1)
+        {
+            Console.Error.WriteLine(errorMessage);
+            Environment.Exit(exitCode);
+        }
+
+        private static void WriteStatusAndExit(ApiStatus apiStatus)
+        {
+            WriteErrorAndExit(apiStatus.ErrorMessage);
+        }
+
+        // TODO: Pull this out to a separate sample.
+        public static void BasicUsageExample(string [] args)
+        {
+            const float outcome = 1.0f;
+            const string eventId = "event_id";
+            const string contextJson = @"({
+                                  ""GUser"":{""id"":""a"",""major"":""eng"",""hobby"":""hiking""},
+                                  ""_multi"":[ { ""TAction"":{""a1"":""f1""} },{""TAction"":{""a2"":""f2""}}]})";
+
+            if (args.Length != 1) {
+                WriteErrorAndExit("Missing path to client configuration json");
+            }
+
+            if (!File.Exists(args[0]))
+            {
+                WriteErrorAndExit($"Could not find file with path '{args[0]}'.");
+            }
+
+            string json = File.ReadAllText(args[0]);
+
+            ApiStatus apiStatus = new ApiStatus();
+
+            Configuration config;
+            if (!Configuration.TryLoadConfigurationFromJson(json, out config, apiStatus))
+            {
+                WriteStatusAndExit(apiStatus);
+            }
+
+            LiveModel liveModel = new LiveModel(config);
+            if (!liveModel.TryInit(apiStatus))
+            {
+                WriteStatusAndExit(apiStatus);
+            }
+
+            RankingResponse rankingResponse = new RankingResponse();
+            if (!liveModel.TryChooseRank(eventId, contextJson, rankingResponse, apiStatus))
+            {
+                WriteStatusAndExit(apiStatus);
+            }
+
+            long actionId;
```

Comment:
Is there any reason why we use `long` instead of `int`?

Reply:
This is due to `int` being defined as 32-bit integer in C#; in vw we use size_t, which is technically uint64 (or rather UIntPtr, which varies depending on platform, but we only target x64 with C# right now). I'm trying to avoid exposing uint64 right now, since these types are not CLSCompliant.

Reply:
got it